### PR TITLE
feat: skip branch unprotect/protect if values are unchanged

### PR DESCRIPTION
**Pull request by YuraBeznos**
Skip branch unprotect/protect if the same values are already set.

This will fix egnyte#178

### CODE REVIEW

**Comment by gdubicki** [comment]
Thanks for your contribution, @YuraBeznos ! :) Please check and fix, if necessary, the failing tests.

File: gitlabform/gitlabform/processors/util/branch_protector.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ def protect_branch(self, project_and_group, configuration, branch):
                     and "merge_access_level"
                     and "unprotect_access_level"
                 ) in configuration["branches"][branch]:
+                    branch_access_levels = self.gitlab.get_branch_access_levels(project_and_group, branch)
+                    levels = ["push_access_levels", "merge_access_levels", "unprotect_access_levels"]
+                    if all(configuration["branches"][branch][level[0:-1]] == branch_access_levels[level][0]["access_level"] for level in levels):
```
Comment (gdubicki) [inline gitlabform/gitlabform/processors/util/branch_protector.py:37]: Erm... What? 😅 
Can you either simplify this code (preferably) or add a comment explaining what is happening here?

Comment (gdubicki) [reply gitlabform/gitlabform/processors/util/branch_protector.py:37]: A friendly reminder, @YuraBeznos. :)

**Comment by gdubicki** [comment]
Hey @YuraBeznos ! Version 2 of GitLabForm will be released with days so it would be great if you could rebase your branch against `v2` and target your PR to that branch. Can you do that please?

File: gitlabform/gitlabform/processors/util/branch_protector.py (hunk 1 above)
Comment (YuraBeznos) [reply gitlabform/gitlabform/processors/util/branch_protector.py:37]: Hey, hello. Sorry for a late response. Looks like I will find time to fix tests somewhere in summer. I do remember about this but slightly busy with other things.

Comment (robin-moss) [reply gitlabform/gitlabform/processors/util/branch_protector.py:37]: Is the following comment enough to explain what the `if all(..)` is doing?
```
# Check each access type has the correct access level, if they do, not change is needed
# Gitlabform uses access_levels with plural form, and gitlab uses singular form.
# [0:-1] removes the plural
```

**Comment by YuraBeznos** [comment]
> Hey @YuraBeznos ! Version 2 of GitLabForm will be released with days so it would be great if you could rebase your branch against `v2` and target your PR to that branch. Can you do that please?

I tried, but looks like made it wrong :)

File: gitlabform/gitlabform/processors/util/branch_protector.py (hunk 1 above)
Comment (gdubicki) [reply gitlabform/gitlabform/processors/util/branch_protector.py]: Yes, that comment is enough, thanks @robin-moss! ...but it's the other way around, it's GitLab that uses plural (see https://docs.gitlab.com/ee/api/protected_branches.html#list-protected-branches) while GitLabForm uses singular.

**Comment by gdubicki** [comment]
> > Hey @YuraBeznos ! Version 2 of GitLabForm will be released with days so it would be great if you could rebase your branch against `v2` and target your PR to that branch. Can you do that please?
> 
> I tried, but looks like made it wrong :)

I think that it will be easier to create a new branch from v2, cherry-pick your commit 3134dfe there and then force-push to the source branch of this PR.

**Comment by YuraBeznos** [comment]
> > > Hey @YuraBeznos ! Version 2 of GitLabForm will be released with days so it would be great if you could rebase your branch against `v2` and target your PR to that branch. Can you do that please?
> > 
> > 
> > I tried, but looks like made it wrong :)
> 
> I think that it will be easier to create a new branch from v2, cherry-pick your commit [3134dfe](https://github.com/egnyte/gitlabform/commit/3134dfe0398581936dc920bda86cc7f82f62fd69) there and then force-push to the source branch of this PR.

Right now should be better.

File: gitlabform/gitlabform/processors/util/branch_protector.py (hunk 1 above)
Comment (YuraBeznos) [reply gitlabform/gitlabform/processors/util/branch_protector.py]: I've updated this also. Thanks.

**Comment by gdubicki** [comment]
LGTM, thanks for your contribution @YuraBeznos !

**Pull request merged**